### ISSUE TITLE
Audioquality low on any Video

### ISSUE THREAD

**Issue by elec2** [TeamNewPipe/NewPipe#11522]
### Checklist

- [X] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.27.2

### Steps to reproduce the bug

1. Open any video
2. Press play
3. Listen to the sound, especially low loudness noise sounds like beach or wind sounds.
4

### Expected behavior

The sound quality should change together with the change of the videoquality.

### Actual behavior

On every quality level, the soundquality is like with 360p video. I think it is 32bit/s Audio.

### Affected Android/Custom ROM version

Android 14

### Affected device model

Samsung Fold 4

**Comment by n-ce** [comment]
Duplicate of https://github.com/TeamNewPipe/NewPipe/issues/8148

**Comment by elec2** [comment]
> Duplicate of #8148

Sorry @n-ce , but it's definitely not the same problem, because some month ago the problem wasn't there.
As the Autor of #8148 says: `This only happens when using data AND in background mode. If I play a video with mobile data the audio is fine. When using WiFi the audio is fine in both cases.`

This is not the case here! It happens in WLAN on mobile and no matter if the app is in background or not.

**Comment by elec2** [comment]
@opusforlife2 no, but while I was testing this again now, I figured a different behavior. Now the sound in WLAN is always good (no matter the video quality settings) and the sound on mobile data is always bad (no matter the settings). This shouldn't be intended right? Because why do I have to rely on a bad audio quality while watching a video at 4k and while being on LTE (especially if I have unlimited data). Is the audio codec automatically selected from youtube or from newpipe?

**Comment by elec2** [comment]
> What about background vs foreground then? No change in audio quality on mobile data?

Exactly

**Comment by AudricV** [comment]
Setting the video resolution limit on mobile data to no limit should resolve the issue, using the lowest audio quality is based on that setting (see the call to `isLimitingDataUsage` in the code [here](https://github.com/TeamNewPipe/NewPipe/blob/035c394cf6704c0af9077c01ebfa55b810d02140/app/src/main/java/org/schabi/newpipe/util/ListHelper.java#L747))

This handling of audio quality based on video resolution limit and whether the network is metered should be completely removed and based on dedicated audio settings.

**Comment by elec2** [comment]
> Setting the video resolution limit on mobile data to no limit should resolve the issue,

Huh, you are right. Thank you!
